app/models/agents/evernote_agent.rb

```diff
@@ -0,0 +1,380 @@
+
+        note = Evernote::EDAM::Type::Note.new(params)
+        updateNote(note)
+        find_note(params[:guid])
+      end
+
+      def find_note(guid)
+        # https://dev.evernote.com/doc/reference/NoteStore.html#Fn_NoteStore_getNote
+        en_note = getNote(guid, true, false, false, false)
+        build_note(en_note)
+      end
+
+      def build_note(en_note)
+        notebook = find_notebook(guid: en_note.notebookGuid).try(:name)
+        tags = en_note.tagNames || find_tags(en_note.tagGuids.to_a).map(&:name)
+        Note.new(en_note, notebook, tags)
+      end
+
+      def find_tags(guids)
+        listTags.select {|tag| guids.include?(tag.guid)}
+      end
+
+      def find_notebook(params)
+        if params[:guid]
+          listNotebooks.detect {|notebook| notebook.guid == params[:guid]}
+        elsif params[:name]
+          listNotebooks.detect {|notebook| notebook.name == params[:name]}
+        end
+      end
+
+      def create_notebook(name)
+        notebook = Evernote::EDAM::Type::Notebook.new(name: name)
+        createNotebook(notebook)
+      end
+
+      def with_wrapped_content(params)
+        params.delete(:notebook)
+
+        if params[:content]
+          params[:content] =
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" \
+            "<!DOCTYPE en-note SYSTEM \"http://xml.evernote.com/pub/enml2.dtd\">" \
+            "<en-note>#{params[:content].encode(:xml => :text)}</en-note>"
+        end
+
+        params
+      end
+
+      class Search
+        attr_reader :note_store, :opts
+        def initialize(note_store, opts)
+          @note_store = note_store
+          @opts = opts
+        end
+
+        def note_guids
+          filtered_metadata.map(&:guid)
+        end
+
+        def notes
+          metadata = filtered_metadata
+
+          if opts[:last_checked_at] && opts[:tagNames]
+
+            # evernote does note change Note#updated timestamp when a tag is added to a note
+            # the following selects recently updated notes
+            # and notes that recently had the specified tags added
+            metadata.select! do |note_data|
+              note_data.updated > opts[:last_checked_at] ||
+              !opts[:notes_with_tags].include?(note_data.guid)
+            end
+
+          elsif opts[:last_checked_at]
+            metadata.select! { |note_data| note_data.updated > opts[:last_checked_at] }
+          end
+
+          metadata.map! { |note_data| note_store.find_note(note_data.guid) }
+          metadata
+        end
+
+        def create_filter
+          filter = Evernote::EDAM::NoteStore::NoteFilter.new
+
+          # evernote search grammar:
+          # https://dev.evernote.com/doc/articles/search_grammar.php#Search_Terms
+          query_terms = []
+          query_terms << "notebook:\"#{opts[:notebook]}\""   if opts[:notebook].present?
+          query_terms << "intitle:\"#{opts[:title]}\""       if opts[:title].present?
```

Comment:
I just tested it and evernote does escape quotes, at least. I'm not sure if anything else would need to be escaped.